pyrdp/recording/recorder.py

```diff
@@ -86,17 +86,30 @@ class FileLayer(LayerChainItem):
     Layer that saves RDP events to a file for later replay.
     """
 
+    FLUSH_THRESHOLD = 18
+
     def __init__(self, fileName: Union[str, Path]):
         super().__init__()
-        self.file_descriptor = open(str(fileName), "wb")
+
+        # Buffer data until we have enough bytes for a meaningful replay.
+        self.pending = b''
+
+        self.filename = fileName
+        self.fd = None
```

Comment:
I wouldve kept `fileDescriptor` because

>Explicit is better than implicit.

and
>Readability counts.

Reply:
I would argue that `fd` is very explicit and readable as the linux API and manpages are always referring to `fd`s in all of its signatures for files, and it's very common to have the constructor as `with open(x) as f:` or `as fd:` The class also fits in a screenful...

Reply:
Fair enough. However, to align with the rest of the codebase, I'd put the variable type (`io.TextIOWrapper`) because the variable is instanciated to None, which removes completion capability.

Reply:
I'm pretty sure `TextIOWrapper` is not the right type, since this is a binary file. I've added type hinting for BufferedIOBase, which exports the right `write()` signature.

Reply:
My bad, I looked to quickly. BufferedIOBase looks good indeed!